app/models/fe/page.rb

```diff
@@ -62,7 +66,18 @@ def questions_before_position(position)
 
     # Include nested elements
     def all_elements
-      (elements + elements.collect(&:all_elements)).flatten
+      if all_element_ids.nil?
+        rebuild_all_element_ids
+        all_elements
+      else
+        ids = all_element_ids.split(',')
+        ids << 0 # in case there are no elements, the query breaks at IN ()
+        Element.where(id: ids).order(:position)
+      end
```

Comment:
This if/else can be replaced with:
```ruby
rebuild_all_element_ids if all_element_ids.nil?
ids = all_element_ids.split(',')
ids.present? ? Element.where(id: ids).order(:position) : Element.where("1 = 0")
```
A few advantages:
1) A bit more compact.
2) Removes risk of infinite loop.
3) That last bit for handling no elements is mostly personal preferences. The code is clear without a comment, which is always nice.